homework11/hw1.py

```diff
@@ -0,0 +1,54 @@
+"""
+Vasya implemented nonoptimal Enum classes.
+Remove duplications in variables declarations using metaclasses.
+
+from enum import Enum
+
+
+class ColorsEnum(Enum):
+    RED = "RED"
+    BLUE = "BLUE"
+    ORANGE = "ORANGE"
+    BLACK = "BLACK"
+
+
+class SizesEnum(Enum):
+    XL = "XL"
+    L = "L"
+    M = "M"
+    S = "S"
+    XS = "XS"
+
+
+Should become:
+
+class ColorsEnum(metaclass=SimplifiedEnum):
+    __keys = ("RED", "BLUE", "ORANGE", "BLACK")
+
+
+class SizesEnum(metaclass=SimplifiedEnum):
+    __keys = ("XL", "L", "M", "S", "XS")
+
+
+assert ColorsEnum.RED == "RED"
+assert SizesEnum.XL == "XL"
+"""
+
+
+class SimplifiedEnum(type):
+    """Metaclass to store variables passed in in __key attribute of instance-class
+    class ColorsEnum(metaclass=SimplifiedEnum):
+        __keys = ("RED", "BLUE", "ORANGE", "BLACK")
+
+    assert ColorsEnum.RED == "RED"
+    assert ColorsEnum.ORANGE == "ORANGE"
+    """
+
+    def __init__(cls, name, bases, dct):
+        super(SimplifiedEnum, cls).__init__(name, bases, dct)
+
+    def __getattr__(cls, item):
+        if item in cls.__dict__[f"_{cls.__name__}__keys"]:
+            return item
+        else:
+            raise KeyError("This key is not declared in the class's __key attribute")
```

Comment:
It is better not to override the `__getattr__` magic method for classes, but to set attributes from the` __keys` list when creating a class using the `__new__` magic method.